File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {
    static boolean[] visited;
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        visited = new boolean[computers.length];
        
        for(int i = 0; i < computers.length; i++){
            if(!visited[i]){
                dfs(i, computers);
                answer++;
            }
        }
        return answer;
    }
    
    public void dfs(int start, int[][] computers){
        visited[start] = true;
        for(int i = 0; i < computers[start].length; i++){
            int next = i;
            if(computers[start][i] == 0)
                continue;
            System.out.println(next);
            if(!visited[next]){
                dfs(next, computers);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;
vector<int> graph[201];
bool visited[201];

void dfs(int start){
    visited[start] = true;
    
    for(int i = 0; i < graph[start].size(); i++){
        int next = graph[start][i];
        if(visited[next])
            continue;
        dfs(next);
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for(int i = 0; i < computers.size(); i++){
        for(int j = 0; j < computers[i].size(); j++){
            if(i == j || computers[i][j] == 0)
                continue;
            graph[i].push_back(j);
        }
    }

    for(int i = 0; i < n; i++){
        if(!visited[i]){
            dfs(i);
            answer++;
        }
    }
    
    return answer;
}